helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -588,20 +596,16 @@ public void resetPartition(String clusterName, String instanceName, String resou
 
     // check partition exists in resource group
     Set<String> resetPartitionNames = new HashSet<String>(partitionNames);
+    Set<String> partitions;
     if (idealState.getRebalanceMode() == RebalanceMode.CUSTOMIZED) {
-      Set<String> partitions = new HashSet<String>(idealState.getRecord().getMapFields().keySet());
-      if (!partitions.containsAll(resetPartitionNames)) {
-        throw new HelixException(
-            "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-                + ", because not all " + partitionNames + " exist");
-      }
+       partitions = new HashSet<String>(idealState.getRecord().getMapFields().keySet());
```

Comment:
Suggest diamond `new HashSet<>()`
And it seems the partitions are only used for next step `containsAll()`. Would it be fine just use the reference to keySet() `partitions = Address unclean helix controller leadership switch`, instead of making a copy?

Reply:
TFTR. Will fix.